ouroboros-consensus/src-unix/Ouroboros/Storage/IO.hs

```diff
@@ -79,6 +80,15 @@ write :: FHandle -> Ptr Word8 -> Int64 -> IO Word32
 write h data' bytes = withOpenHandle "write" h $ \fd ->
     fromIntegral <$> Posix.fdWriteBuf fd data' (fromIntegral bytes)
 
+-- Like write, but it also makes sure that the whole buffer is flushed.
+writeFull :: FHandle -> Ptr Word8 -> Int64 -> IO ()
+writeFull h data' bytes =
+    let go _ _ 0 = return ()
```

Comment:
IMO, using `where`here instead of `let` would be more readable and also make it more evident that the local function doesn't depend on any bindings except for the function arguments.